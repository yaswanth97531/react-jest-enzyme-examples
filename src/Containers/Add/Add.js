import React, { Component } from 'react';
import Form from './../Form/Form';

class Add extends Component {
  render() {
    return (
      <div className="App">
        <h1>Add Function</h1>
        <p>I don't care</p>
        <Form operator='-' />
      </div>
    );
  }
}

export default Add;