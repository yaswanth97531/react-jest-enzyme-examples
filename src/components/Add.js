import React, { Component } from 'react';
import Form from './form';

class Add extends Component {
  render() {
    return (
      <div className="App">
        <h1>Add Function</h1>
        <p>I don't care</p>
        <Form operator='+' />
      </div>
    );
  }
}

export default Add;