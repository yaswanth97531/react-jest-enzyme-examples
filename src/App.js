import React from 'react';

import './App.css';
import Add from './Containers/Add/Add';
import CounterList from './Containers/CounterList/CounterList';
import List from './Containers/ItemsList/List';
import Calculator from './Containers/Calculator/Calculator';
import DataTable from './Containers/DataTable/DataTable';

const cols = [{
    header: 'ID',
    name: 'id'
  },
  {
    header: 'Name',
    name: 'name'
  },
  {
    header: 'Email',
    name: 'email'
  }
];
const data = [{
    id: 5,
    name: 'John',
    email: 'john@example.com'
  },
  {
    id: 6,
    name: 'Liam',
    email: 'liam@example.com'
  },
  {
    id: 7,
    name: 'Maya',
    email: 'maya@example.com',
    someTest: 10
  },
  {
    id: 8,
    name: 'Oliver',
    email: 'oliver@example.com',
    hello: 'hello world'
  },
  {
    id: 25,
    name: 'Amelia',
    email: 'amelia@example.com'
  }
];

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <Add />
        <hr/>
        <CounterList />
        <hr/>
        {/* <List items={['one', 'two', 'three']} /> */}
        <List items={[]} />
        <hr />
        <Calculator />
        <hr/>
        {/* <DataTable cols={cols} data={data} /> */}
        <DataTable cols={[]} data={[]} />
      </header>
    </div>
  );
}

export default App;
