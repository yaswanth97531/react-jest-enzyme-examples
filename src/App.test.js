import App from './App';
import Add from './Containers/Add/Add'
import Calculator from './Containers/Calculator/Calculator';

let wrapper;

beforeEach(() => {
  wrapper = mount( <App /> );
});

describe('<App /> rendering', () => {
  it('should render correctly', () => {
    const tree = renderer.create( < App / > ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render the \'Add\' Component', () => {
    expect(wrapper.containsMatchingElement(<Add />)).toEqual(true);
  });

  it('should render the \'Calculator\' Component', () => {
    expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true);
  });
});
describe('<App /> interactions', () => {});
describe('<App /> lifecycle method invocations', () => {});