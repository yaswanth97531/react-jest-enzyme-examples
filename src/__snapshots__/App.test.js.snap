// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> rendering should render correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="App"
    >
      <h1>
        Add Function
      </h1>
      <p>
        I don't care
      </p>
      <form
        className="form-group"
      >
        <fieldset
          className="form-group"
        >
          <label
            className="form-label"
          >
            1st Number:
          </label>
          <input
            className="form-input"
            id="number1"
            onChange={[Function]}
            type="text"
            value=""
          />
        </fieldset>
        <fieldset
          className="form-group"
        >
          <label
            className="form-label"
          >
            2nd Number:
          </label>
          <input
            className="form-input"
            id="number2"
            onChange={[Function]}
            type="text"
            value=""
          />
        </fieldset>
        <div
          className="form-group"
        >
          <button
            className="btn"
            id="formButtonSubtract"
            onClick={[Function]}
            type="button"
          >
            Subtract
          </button>
        </div>
      </form>
    </div>
    <hr />
    <div>
      <button
        onClick={[Function]}
      >
        Add Counter
      </button>
      <ul>
        <li>
          <div>
            <p>
              Current count: 
              0
            </p>
            <button
              id="increment"
              onClick={[Function]}
            >
              +++
            </button>
            <button
              id="decrement"
              onClick={[Function]}
            >
              ---
            </button>
          </div>
        </li>
        <li>
          <div>
            <p>
              Current count: 
              0
            </p>
            <button
              id="increment"
              onClick={[Function]}
            >
              +++
            </button>
            <button
              id="decrement"
              onClick={[Function]}
            >
              ---
            </button>
          </div>
        </li>
      </ul>
    </div>
    <hr />
    <span
      id="empty-message"
    >
      No items in list
    </span>
    <hr />
    <div
      className="calculator-container"
    >
      <div
        className="display-container"
      >
        <p
          className="display-value"
        >
          0
        </p>
      </div>
      <div
        className="keypad-container"
      >
        <div
          className="numbers-container"
        >
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              9
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              8
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              7
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              6
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              5
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              4
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              3
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              2
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              1
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              .
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              0
            </p>
          </div>
          <div
            className="key-container number-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              ce
            </p>
          </div>
        </div>
        <div
          className="operators-container"
        >
          <div
            className="key-container operator-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              /
            </p>
          </div>
          <div
            className="key-container operator-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              X
            </p>
          </div>
          <div
            className="key-container operator-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              -
            </p>
          </div>
          <div
            className="key-container operator-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              +
            </p>
          </div>
        </div>
        <div
          className="submit-container"
        >
          <div
            className="key-container submit-key"
            onClick={[Function]}
          >
            <p
              className="key-value"
            >
              =
            </p>
          </div>
        </div>
      </div>
    </div>
    <hr />
    <table>
      <thead>
        <tr />
      </thead>
      <tbody>
        <tr>
          <td
            colSpan={0}
          >
            There is no data in this table
          </td>
        </tr>
      </tbody>
    </table>
  </header>
</div>
`;
